campaigns/sync.py

```
@@ -0,0 +1,399 @@
+            if not campaign:
+                return Response({"error": "Campaign not found on chain"}, status=404)
+
+            return Response(
+                {
+                    "success": True,
+                    "message": "Campaign created" if created else "Campaign updated",
+                    "on_chain_id": campaign.on_chain_id,
+                }
+            )
+
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)